### PR TITLE
multiple extension handler

### PR DESCRIPTION
**Pull request by tvelu77**
the program recognizes PlantUML, mermaid and markdown.

We just need to implement PlantCmdLineParser and some refactoring in Editor class

### CODE REVIEW

File: src/main/java/com/github/forax/umldoc/Main.java [hunk 1]
```diff
@@ -33,35 +33,43 @@ public static void main(String[] args) {
...
+      System.err.println("Couldn't find a Module");
       return;
     }
     List<Package> packages;
     try {
-      packages = ModuleScrapper.scrapModule(module.get());
+      packages = ModuleScrapper.scrapModule(modules.next());
     } catch (IOException e) {
       System.err.println("Couldn't get the list of Packages : " + e.getMessage());
       return;
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/Main.java:47]: call System.exit() with a non zero code

File: src/main/java/com/github/forax/umldoc/Main.java [hunk 2]
```diff
@@ -33,35 +33,43 @@ public static void main(String[] args) {
...
     }
 
+    var index = args[1].lastIndexOf(".");
+    if (index == -1) {
+      System.err.println("Couldn't find an extension in " + args[1]);
+      return;
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/Main.java:53]: same here

File: src/main/java/com/github/forax/umldoc/editor/Editor.java [hunk 3]
```diff
@@ -22,22 +22,39 @@ enum State {
     READWRITE
   }
 
+  /**
+   * The enum Extension corresponds to the type of the file.
+   * <br>Mermaid, PlantUML and Markdown.
+   */
+  enum Extension {
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/editor/Editor.java:29]: should be public

File: src/main/java/com/github/forax/umldoc/editor/Editor.java [hunk 4]
```diff
@@ -22,22 +22,39 @@ enum State {
...
+  private final Extension extension;
 
   /**
    * The constructor of Editor.
    *
    * @param registration Map, register a String with a CommandLineParser.
    * @param module List, a list of package.
    */
-  public Editor(Map<String, CommandLineParser> registration,
+  public Editor(String extension, Map<String, CommandLineParser> registration,
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/editor/Editor.java:47]: String is a poor man object, the extension should be typed as Extension and the switch should be in the calling code

File: src/main/java/com/github/forax/umldoc/Main.java (hunk 1 above)
Comment (tvelu77) [reply src/main/java/com/github/forax/umldoc/Main.java:47]: According to the javadoc, -1 is the value to use when the program is not successful.

Should the exit method take as a parameter : -1 or another value ?

Comment (forax) [reply src/main/java/com/github/forax/umldoc/Main.java]: -1 is great

File: src/main/java/com/github/forax/umldoc/Main.java [hunk 5]
```diff
@@ -30,38 +33,58 @@ public static void main(String[] args) {
...
     var finder = ModuleFinder.of(Path.of(args[0]));
-    var module = finder.find("com.github.forax.umldoc");
+    var modules = finder.findAll().iterator();
 
-    if (module.isEmpty()) {
-      System.err.println("Couldn't find the Module");
-      return;
+    if (!modules.hasNext()) {
+      System.err.println("Couldn't find a Module");
+      System.exit(-1);
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/Main.java:43]: System.exit() can fail (see the javadoc) so it should be followed by a return

File: src/main/java/com/github/forax/umldoc/Main.java [hunk 6]
```diff
@@ -30,38 +33,62 @@ public static void main(String[] args) {
...
 
-    if (module.isEmpty()) {
-      System.err.println("Couldn't find the Module");
+    if (!modules.hasNext()) {
+      System.err.println("Couldn't find a Module");
+      System.exit(-1);
       return;
     }
-    List<Package> packages;
+    var packages = List.<Package>of();
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/Main.java:47]: you do not need it anymore

File: src/main/java/com/github/forax/umldoc/editor/Editor.java [hunk 7]
```diff
@@ -48,6 +64,8 @@ public Editor(Map<String, CommandLineParser> registration,
    * @throws IOException If the file cannot be opened.
    */
   public void edit(Writer writer, BufferedReader reader) throws IOException {
+    Objects.requireNonNull(writer);
+    Objects.requireNonNull(reader);
     var line = "";
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/editor/Editor.java:69]: this initialization is not necessary

**Pull request merged**